generators/upgrade/index.js

```diff
@@ -79,7 +79,8 @@ module.exports = UpgradeGenerator.extend({
         this.log(`Regenerating application with JHipster ${version}...`);
         let generatorCommand = 'yo jhipster';
         if (semver.gte(version, FIRST_CLI_SUPPORTED_VERSION)) {
-            generatorCommand = this.clientPackageManager === 'yarn' ? '$(yarn bin)/jhipster' : '$(npm bin)/jhipster';
+            const generatorDir = this.clientPackageManager === 'yarn' ? shelljs.exec('yarn bin', { silent: true }).stdout : shelljs.exec('npm bin', { silent: true }).stdout;
```

Comment:
Please change ` { silent: true }` to ` { silent: this.silent }` so that the `--verbose` flag will print the result of the command (this is how the rest of the shelljs commands in this file are set up)